core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -0,0 +1,529 @@
+    List<Acl> aclsFromSOT;
+    if (!regex) {
+      aclsFromSOT =
+          manageDatabase.getHandleDbRequests().getSyncAcls(env, topicNameSearch, tenantId);
+    } else {
+      aclsFromSOT = manageDatabase.getHandleDbRequests().getSyncAcls(env, tenantId);
+      List<Acl> topicFilteredList = aclsFromSOT;
+      // Filter topics on topic name for search
+      if (topicNameSearch != null && topicNameSearch.length() > 0) {
+        final String topicSearchFilter = topicNameSearch;
+        topicFilteredList =
+            aclsFromSOT.stream()
+                .filter(acl -> acl.getTopicname().contains(topicSearchFilter))
+                .collect(Collectors.toList());
+      }
+      aclsFromSOT = topicFilteredList;
+    }
+
+    return aclsFromSOT;
+  }
+
+  private Map<String, String> getTopicPromotionEnv(
+      String topicSearch, List<Topic> topics, int tenantId) {
+    Map<String, String> hashMap = new HashMap<>();
+    try {
+      if (topics == null) {
+        topics = manageDatabase.getHandleDbRequests().getTopics(topicSearch, tenantId);
+      }
+
+      hashMap.put("topicName", topicSearch);
+
+      if (topics != null && topics.size() > 0) {
+        List<String> envList = new ArrayList<>();
+        topics.forEach(topic -> envList.add(topic.getEnvironment()));
+
+        // tenant filtering
+        String orderOfEnvs = mailService.getEnvProperty(tenantId, "ORDER_OF_ENVS");
+        envList.sort(Comparator.comparingInt(orderOfEnvs::indexOf));
+
+        String lastEnv = envList.get(envList.size() - 1);
+        List<String> orderdEnvs = Arrays.asList(orderOfEnvs.split(","));
+
+        if (orderdEnvs.indexOf(lastEnv) == orderdEnvs.size() - 1) {
+          hashMap.put("status", "NO_PROMOTION"); // PRD
+        } else {
+          hashMap.put("status", ApiResultStatus.SUCCESS.value);
+          hashMap.put("sourceEnv", lastEnv);
+          String targetEnv = orderdEnvs.get(orderdEnvs.indexOf(lastEnv) + 1);
+          hashMap.put("targetEnv", getEnvDetails(targetEnv, tenantId).getName());
+          hashMap.put("targetEnvId", targetEnv);
+        }
+
+        return hashMap;
+      }
+    } catch (Exception e) {
+      log.error("getTopicPromotionEnv error ", e);
+      hashMap.put("status", ApiResultStatus.FAILURE.value);
+      hashMap.put("error", "Topic does not exist in any environment.");
+    }
+
+    return hashMap;
+  }
+
+  private List<Topic> getFilteredTopicsForTenant(List<Topic> topicsFromSOT) {
+    // tenant filtering
+    try {
+      List<String> allowedEnvIdList = getEnvsFromUserId(getUserName());
+      if (topicsFromSOT != null) {
+        topicsFromSOT =
+            topicsFromSOT.stream()
+                .filter(topic -> allowedEnvIdList.contains(topic.getEnvironment()))
+                .collect(Collectors.toList());
+      }
+    } catch (Exception exception) {
+      log.error("No environments/clusters found.", exception);
+      return new ArrayList<>();
```

Comment:
Looks like this will fail.
I have looked for usages and found this line
```java
Integer topicOwnerTeam = getFilteredTopicsForTenant(topicsSearchList).get(0).getTeamId();
```
it means it expects at least one element in the result. And it will fail for empty list